slack_sdk/web/async_client.py

```diff
@@ -2992,6 +2995,160 @@ async def files_upload(
+        * step2: "https://files.slack.com/upload/v1/..." URLs returned from files.getUploadURLExternal API
+        * step3: https://api.slack.com/methods/files.completeUploadExternal
+        """
+        if file is None and content is None and upload_files is None:
+            raise e.SlackRequestError("Any of file, content, and upload_files must be specified.")
+        if file is not None and content is not None:
+            raise e.SlackRequestError("You cannot specify both the file and the content argument.")
+        if channels is not None and (
+            (isinstance(channels, (list, Tuple)) and len(channels) > 1)
+            or (isinstance(channels, str) and len(channels.split(",")) > 1)
+        ):
+            raise e.SlackRequestError("Sharing files with multiple channels is no longer supported")
+        if filetype is not None:
+            warnings.warn("filetype is no longer supported. Please remove it from the arguments.")
+
+        # step1: files.getUploadURLExternal per file
+        files: List[Dict[str, Any]] = []
+        if upload_files is not None:
+            for f in upload_files:
+                files.append(_to_v2_upload_file_item(f))
+        else:
+            f = _to_v2_upload_file_item(
+                {
+                    "filename": filename,
+                    "file": file,
+                    "content": content,
+                    "title": title,
+                    "alt_txt": alt_txt,
+                    "snippet_type": snippet_type,
+                }
+            )
+            files.append(f)
+
+        for f in files:
+            url_response = await self.files_getUploadURLExternal(
+                filename=f.get("filename"),
+                length=f.get("length"),
+                alt_txt=f.get("alt_txt"),
+                snippet_type=f.get("snippet_type"),
+                token=kwargs.get("token"),
+            )
+            f["file_id"] = url_response.get("file_id")
+            f["upload_url"] = url_response.get("upload_url")
+
+        # step2: "https://files.slack.com/upload/v1/..." per file
+        for f in files:
+            upload_result = _upload_file_via_v2_url(
+                url=f["upload_url"],
+                data=f["data"],
+                logger=self._logger,
+                timeout=self.timeout,
+                proxy=self.proxy,
+                ssl=self.ssl,
+            )
+            if upload_result.get("status") != 200:
+                status = upload_result.get("status")
+                body = upload_result.get("body")
+                message = (
+                    "Failed to upload a file "
+                    f"(status: {status}, body: {body}, filename: {f.get('filename')}, title: {f.get('title')})"
+                )
+                raise e.SlackRequestError(message)
+
+        # step3: files.completeUploadExternal with all the sets of (file_id + title)
+        channel_to_share = channel
+        if channels is not None:
```

Comment:
If having n channels is no longer supported do we want to include this logic?
Maybe we could send a more relevant error message above?
```python
raise e.SlackRequestError("Sharing files with multiple channels is no longer supported, share files in each channel separately")
```

Reply:
thanks - i will use this error message but still would like to support single channel item use cases with warning